etl::optional issues -Wmaybe-uninitialized warnings under some circumstances

Hi! First of all thanks for this great project!

I encountered an issue with `etl::optional`. Under some circumstances the constructor `etl::optional(etl::nullopt_t)` produces a `-Wmaybe-uninitialized` warning when compiling with gcc. I wrote the following MWE:
```c++
// repro.cpp

#include "etl/optional.h"
#include "etl/string.h"
#include "etl/to_string.h"

class Repro {
public:
    void doSomething(etl::optional<unsigned> value = etl::nullopt) {
        _buffer.clear();
        if (value) {
            etl::to_string(*value, _buffer, true);
        }
    }
private:
    etl::string<20> _buffer{};
};

int main(int , char *[])
{
    Repro repro;
    repro.doSomething();
    return 0;
}
``` 
When compiling this file with `arm-none-eabi-gcc` (I used version 11.1.0) with `-Og`, I get the following and rather cryptic diagnostic:
```
$ arm-none-eabi-gcc -Ietl/include -Wall -Wextra -Og -std=c++17 -xc++ repro.cpp       

In member function 'void Repro::doSomething(etl::optional<unsigned int>)',
    inlined from 'int main(int, char**)' at repro.cpp:20:22:
repro.cpp:10:27: warning: '*(unsigned int*)((char*)&<unnamed> + offsetof(etl::optional<unsigned int>,etl::optional<unsigned int>::storage.etl::aligned_storage<4, 4>::type::data[0]))' may be used uninitialized [-Wmaybe-uninitialized]
   10 |             etl::to_string(*value, _buffer, true);
      |             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
repro.cpp: In function 'int main(int, char**)':
repro.cpp:20:22: note: '<anonymous>' declared here
   20 |     repro.doSomething();
      |     ~~~~~~~~~~~~~~~~~^~
``` 
I think this is due to the constructor of `etl::optional(etl::nullopt_t)` from `etl::nullopt` that does not initialise the `storage` member. If I add an initialiser there, the warning is gone. This only happens when compiling with `-Og` so I suspect it depends on how the library code is inlined differently depending on the optimisation level.

--- a/include/etl/optional.h
+++ b/include/etl/optional.h
@@ -112,7 +112,8 @@ namespace etl
     /// Constructor.
     //***************************************************************************
     optional()
-      : valid(false)
+      : storage()
+      , valid(false)
     {
     }
 
@@ -120,7 +121,8 @@ namespace etl
     /// Constructor with nullopt.
     //***************************************************************************
     optional(etl::nullopt_t)
-      : valid(false)
+      : storage()
+      , valid(false)
     {
     }
 

